ccdproc/astrometry.py

```diff
@@ -0,0 +1,478 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# This module implements the combiner class.
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import itertools as it
+import numpy as np
+
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy import wcs
+from astropy.modeling import models
+from astropy.modeling import fitting
+
+__all__ = ['remove_duplicates', 'distance', 'distance_ratios',
+           'triangle_angle', 'calc_triangle', 'match_by_triangle',
+           'match_by_fit', 'create_wcs_from_fit']
+
+
+def remove_duplicates(x, y, tolerance):
+    """Remove duplicates within a certain tolerance from a pair of coordiantes.
+
+    Parameters
+    ----------
+    x : `numpy.ndarray`
+        x-position of objects
+
+    y : `numpy.ndarray`
+        y-position of objects
+
+    tolerance : float
+        tolerance for removal of dupliactes
+
+    Returns
+    -------
+    x : `numpy.ndarray`
+        x-position of objects with duplicates removed
+
+    y : `numpy.ndarray`
+        y-position of objects with duplicates removed
+    """
+    mask = np.ones(len(x), dtype=bool)
+
+    for i in range(len(x)):
+        if mask[i]:
+            d = ((x - x[i])**2 + (y - y[i])**2)**0.5
+            j = np.where((d < tolerance))
+            mask[j] = False
+            mask[i] = True
+
+    return x[mask], y[mask]
+
+
+def distance(x1, y1, x2, y2):
+    """Calcluate the distance between points.
+
+    Parameters
+    ----------
+    x1 : float or `numpy.ndarray`
+        x-position of objects
+
+    y1 : float or `numpy.ndarray`
+        y-position of objects
+
+    x2 : float or `numpy.ndarray`
+        x-position of objects
+
+    y2 : float or `numpy.ndarray`
+        y-position of objects
+
+    Returns
+    -------
+    d : float or `numpy.ndarray`
+        Distance between x1,y1 and x2,y2
+    """
+    return ((x1-x2)**2 + (y1-y2)**2)**0.5
+
+
+def _get_index(z, n_stars):
```

Comment:
I haven't found a place where this function is actually used. Is there a reason to keep it?

Reply:
Used in an earlier version of the code -- other than still might be useful, but if it really isn't needed in the end, it would be good to remove it.

Reply:
Just in case we do need it, I could provide you a solution that mathematically solves your problem instead of "by iteration". I was actually curious if it was easily possible and implemented it, but because the function wasn't actually used I put it on [StackOverflow](https://stackoverflow.com/a/46132585/5393381).